circuit-overlay/src/commonMain/kotlin/com/slack/circuit/overlay/Overlay.kt

```diff
@@ -166,6 +173,31 @@ public interface Overlay<Result : Any> {
   @Composable public fun Content(navigator: OverlayNavigator<Result>)
 }
 
+/**
+ * An [Overlay] that can be animated in and more interestingly out _after_ it has returned a
+ * [Result]. The [Overlay] is animated in and out by its [enterTransition] and [exitTransition],
+ * [AnimatedContent] is executed with with [AnimatedVisibilityScope] so that child animations can be
+ * coordinated with the overlay's animations.
+ */
+public abstract class AnimatedOverlay<Result : Any>(
```

Comment:
I think we should mark this as an experimental API. I can see this API evolving.

Reply:
Up to you here. I still think we can add signal to what APIs are more stable than others?

I don't mind either way though. If you'd rather keep experimental annotations out of here (understandably), all good with me.

Reply:
yeah I think we're fine without it for now, most of the library would probably fall under that annotation's umbrella if we introduced it heh